### PR TITLE
Api Fetch: Fix metaboxes save request (parse: false)

### PR DESCRIPTION
**Pull request by youknowriad**
Related to #7566 

Not certain yet why the e2e failures are intermittent in travis and consistent locally (without this PR).

Meta boxes save requests are not JSON request, we need to passe `parse: false`

### CODE REVIEW

File: packages/api-fetch/src/index.js [hunk 1]
```diff
@@ -55,7 +58,16 @@ function apiFetch( options ) {
...
+				const invalidJsonError = Promise.reject( {
+					code: 'invalid_json',
+					message: __( 'The response is not a valid JSON response.' ),
+				} );
+
+				if ( ! response.json ) {
+					return invalidJsonError;
+				}
+
+				return response.json
```
Comment (youknowriad) [inline packages/api-fetch/src/index.js]: oups

**Comment by aduth** [comment]
>Meta boxes save requests are not JSON request, we need to passe parse: false

Should we not use `apiFetch` for non-JSON endpoints?

Should we update `apiFetch` to handle the case where JSON is not returned?

Seems like it could be a pain point for developers to need to anticipate this.

**Review by aduth — APPROVED** [review]
There should be test cases verifying each of the added condition paths.

**Comment by youknowriad** [comment]
Good call on the tests, I caught a small bug.

**Comment by youknowriad** [comment]
> Should we not use apiFetch for non-JSON endpoints?

I think it's fine to use it as a utility to avoid specifying nonces and full URLs but granted it's an edge case. (Probably only useful for legacy stuff like metaboxes)

**Comment by youknowriad** [comment]
> Seems like it could be a pain point for developers to need to anticipate this.

With the current implementation, it's pretty obvious actually, you write your function, you notice you have a failed promise with invalid JSON.

**Pull request merged**